packages/core/src/api.ts

```diff
@@ -68,32 +72,51 @@ export class Api {
     return this.#chainProperties
   }
 
+  // Set the hook function to be called when api fetch endpoint.
+  // This hook is throttled and won't be called more than once in 50ms timeframe.
+  onFetching(fetching?: () => void) {
+    this.#apiHooks.fetching = fetching
+  }
+
+  fetching = _.throttle(
```

Comment:
maybe we can have the throttle logic on the handler instead? so this is get called on every fetch. just to make the API less complicated

Reply:
That work as well

Reply:
the main thing about denouncing logic is that it is often we forget to clear the timer at destructor and it could cause issue for example this is used in unit tests and creating many Chopsticks instances